### ISSUE TITLE
Use in production

### ISSUE THREAD

**Issue by arhmnsh** [Gardencoder/react-native-paytabs#2]
Hi,
Is this repo active? Can I use it in my production React Native app (iOS and Android)?
Thank you!

**Comment by RamyHegazy** [comment]
@arhmnsh It is working fine with react-native old version, but you can check this https://github.com/Gardencoder/react-native-paytabs-library, i didn't publish npm dependency yet, so you have to add it as sub module in your app.

**Comment by arhmnsh** [comment]
Thanks for the reply @RamyHegazy.
I'm using RN 0.60.x, will it work fine with it?

**Comment by RamyHegazy** [comment]
> Thanks for the reply @RamyHegazy.
> I'm using RN 0.60.x, will it work fine with it?

It should work fine, let me know if you have any issue

**Comment by arhmnsh** [comment]
<img width="587" alt="image" src="https://user-images.githubusercontent.com/1363724/63755283-02a22400-c8bf-11e9-9685-05a520fdb57c.png">

For using it, can I use the same like this? - https://github.com/Gardencoder/react-native-paytabs#usage

**Comment by RamyHegazy** [comment]
Yes, but you have include the library as a local library, cause I don't publish as NPM depenency yet

**Comment by RamyHegazy** [comment]
Might be by next week, it will be ready with latest updates, will update you this week.

**Comment by arhmnsh** [comment]
Thanks a lot for your quick replies.

I tried installing it using `npm install --save Gardencoder/react-native-paytabs-library`
After installing it, my Xcode project won't compile.

<img width="1542" alt="image" src="https://user-images.githubusercontent.com/1363724/63772371-a1d81300-c8e1-11e9-89c0-8fb221a3bb2f.png">


Am I doing this right?

**Comment by RamyHegazy** [comment]
@arhmnsh yes, make sure the library path is correct, in `package.json` it should be something like this

```
"dependencies": {
    "bar": "file:../foo/bar"
 }
```

After install the dependencies, follow this document
https://github.com/Gardencoder/react-native-paytabs/blob/master/docs/IOS.md
https://github.com/Gardencoder/react-native-paytabs/blob/master/docs/ANDROID.md

**Comment by arhmnsh** [comment]
The depency is actually like this, since I've installed it directly from the repo using command 
`npm install --save https://github.com/Gardencoder/react-native-paytabs-library`:


   ```
"dependencies": {
     "react-native-paytabs-library": "github:Gardencoder/react-native-paytabs-library"`,
     ...
}
```

I'll try linking and get back to you

**Comment by arhmnsh** [comment]
I've followed the steps from here: https://github.com/Gardencoder/react-native-paytabs-library#manual-installation
<img width="845" alt="image" src="https://user-images.githubusercontent.com/1363724/63792171-add5cc00-c905-11e9-9879-404620e153b5.png">

But still getting this error:
<img width="847" alt="image" src="https://user-images.githubusercontent.com/1363724/63792202-bb8b5180-c905-11e9-8616-aad60851dd34.png">

**Comment by RamyHegazy** [comment]
We have additional steps, it is a bridge between our native SDKs for IOS/Android and Js code, so please follow instruction here https://github.com/Gardencoder/react-native-paytabs/blob/master/docs/IOS.md, you have to add Paytabs-IOS framework and resource bundle.

**Comment by arhmnsh** [comment]
It's working fine so far for iOS now. 😀 
But on Android, it's giving these errors for minSdkVersion and project structure.

<img width="1920" alt="image" src="https://user-images.githubusercontent.com/1363724/63834506-1a8cad00-c97e-11e9-95a3-77e5babca9f4.png">

**Comment by RamyHegazy** [comment]
@arhmnsh Cool, for android `minSdkVersion 16`, i think it is fine if you are going to publish with play store, but if you have a different requirements you can send https://www.paytabs.com/en/support/

**Comment by RamyHegazy** [comment]
@arhmnsh you can update the minSdkVersion in your android generated project to 16

**Comment by arhmnsh** [comment]
Actually, to clarify, android project doesn't compile.
From the logs, it looks like its complaining about project structure of `react-native-playtabs-library`.

minSdkVersion of my Android is already 16, but looks like minSdkVersion of rn-paytabs is 17.

**Comment by RamyHegazy** [comment]
please check this one: https://github.com/Gardencoder/react-native-paytabs/blob/master/docs/ANDROID.md 

instead of using ` include ':paytabs-android-SDK-v3.0.3'` instead of `':paytabs-android-SDK-v2.5'`

you can find the SDKs versions here: https://github.com/paytabscom/paytabs-android-library-sample/tree/master/sdk

**Comment by arhmnsh** [comment]
Hi @RamyHegazy,
I've tried multiple times to reinstall both the packages. It doesn't work.
I'm getting below errors when running the project in Android Studio.

![image](https://user-images.githubusercontent.com/1363724/64078056-fd8a0e00-ccde-11e9-8e33-ced21774c021.png)


I've added the v3.0.3 SDK like this
<img width="1363" alt="image" src="https://user-images.githubusercontent.com/1363724/64077345-6c179d80-ccd8-11e9-9325-102ea80051f3.png">
<img width="1290" alt="image" src="https://user-images.githubusercontent.com/1363724/64078072-29a58f00-ccdf-11e9-9ce7-1e6b628b623e.png">


I really appreciate your help to solve this, as it is very important for me to integrate PayTabs for Android.
Thank you!

**Comment by RamyHegazy** [comment]
@arhmnsh which repository you are using?, you mention before `github:Gardencoder/react-native-paytabs-library`, in that case it is already with SDK `3.0.2`

**Comment by arhmnsh** [comment]
Yes, I'm using `github:Gardencoder/react-native-paytabs-library`, but then you asked me to also follow the instructions from here https://github.com/Gardencoder/react-native-paytabs/blob/master/docs/ANDROID.md

So i've also installed `rn-paytabs`. this is not necessary?

https://github.com/Gardencoder/react-native-paytabs/issues/2#issuecomment-525707684

**Comment by RamyHegazy** [comment]
> So i've also installed `rn-paytabs`. this is not necessary?
Not necessary at all, it is old repo

**Comment by RamyHegazy** [comment]
@arhmnsh 
## Follow steps to run ANDROID

- Run terminal `npm install rn-paytabs --save`
- Run terminal `react-native link rn-paytabs`
- Append the following lines to `android/settings.gradle`:
  	```
  	include ':rn-paytabs'
  	project(':rn-paytabs').projectDir = new File(rootProject.projectDir, 	'../node_modules/rn-paytabs/android')
	include ':paytabs-android-SDK-v3.0.2'
    project(':paytabs-android-SDK-v3.0.2').projectDir = new File(rootProject.projectDir,  '../node_modules/rn-paytabs/android/paytabs-android-SDK-v3.0.2')
  	```
- If you have issue with okhttp or okio modules:
    ```
	configurations {
        all*.exclude group: 'com.squareup.okhttp3', module: 'okhttp'
        all*.exclude group: 'com.squareup.okio', module: 'okio'
    }
	```
- Enjoy!

**Comment by arhmnsh** [comment]
Okay, I've removed all the PayTabs dependencies
Android project works fine

Now, freshly installed `github:Gardencoder/react-native-paytabs-library`
Linked it properly
Now I get this error `error: package com.reactlibrary does not exist
`
<img width="1853" alt="image" src="https://user-images.githubusercontent.com/1363724/64108521-aa1ecb00-cd85-11e9-83ea-31ce1db7d8c7.png">

**Comment by RamyHegazy** [comment]
Insert the following lines inside the dependencies block in `android/app/build.gradle`:

`  compile project(':react-native-paytabs-library')`

It should have the following dependences:
```
dependencies {
    ...
    implementation 'com.facebook.react:react-native:+'
    implementation project(':paytabs_sdk-v3.0.2')
    implementation 'com.android.support:design:28.0.0'
    implementation 'com.android.support:appcompat-v7:28.0.0'
    implementation 'com.squareup.retrofit2:retrofit:2.4.0'
    implementation 'com.google.code.gson:gson:2.8.5'
    implementation 'com.squareup.retrofit2:converter-gson:2.4.0'
    ...
}
```

**Comment by arhmnsh** [comment]
I've inserted it
<img width="519" alt="image" src="https://user-images.githubusercontent.com/1363724/64111341-1c46de00-cd8d-11e9-807d-448d3261cb6f.png">

Now I get a new warning to not use 'implementation' instead of 'compile'. And the older errors remain same.
<img width="1308" alt="image" src="https://user-images.githubusercontent.com/1363724/64111528-a2fbbb00-cd8d-11e9-9fdc-9ee8dba72aa4.png">

**Comment by RamyHegazy** [comment]
Please remove compile project line, rebuild the project

**Comment by arhmnsh** [comment]
Removed it, still get the error `error: package com.reactlibrary does not exist`

<img width="1379" alt="image" src="https://user-images.githubusercontent.com/1363724/64111915-7d22e600-cd8e-11e9-9957-74205e4bd880.png">

**Comment by RamyHegazy** [comment]
@arhmnsh i would recommend to download the library, it is already has a demo sample, can you please check it first.

**Comment by arhmnsh** [comment]
@RamyHegazy I'll try that and get back to you

**Comment by arhmnsh** [comment]
@RamyHegazy meanwhile, before I was using `rn-paytabs`, so it was working fine on iOS.
But since I removed `rn-tabs`, I'm importing it like `import RNPaytabsLibrary from 'react-native-paytabs-library';`

Now, I get error for merchant_email, even though it was working fine before.
import RNPaytabsLibrary from 'react-native-paytabs-library';

**Comment by RamyHegazy** [comment]
Please share with  a screenshot

**Comment by RamyHegazy** [comment]
Please check the sample here: https://github.com/Gardencoder/react-native-paytabs-library/blob/master/example/App.js

```JS
RNPaytabsLibrary.start({
      [RNPaytabsLibrary.merchant_email]: "rhegazy@paytabs.com",
      [RNPaytabsLibrary.secret_key]: "BIueZNfPLblJnMmPYARDEoP5x1WqseI3XciX0yNLJ8v7URXTrOw6dmbKn8bQnTUk6ch6L5SudnC8fz2HozNBVZlj7w9uq4Pwg7D1",// Add your Secret Key Here
      [RNPaytabsLibrary.transaction_title]: "Mr. John Doe",
      [RNPaytabsLibrary.amount]: "2.0",
      [RNPaytabsLibrary.currency_code]: "USD",
      [RNPaytabsLibrary.customer_email]: "test@example.com",
      [RNPaytabsLibrary.customer_phone_number]: "+97333109781",
      [RNPaytabsLibrary.order_id]: "1234567",
      [RNPaytabsLibrary.product_name]: "Tomato",
      [RNPaytabsLibrary.timeout_in_seconds]: "300", //Optional
      [RNPaytabsLibrary.address_billing]: "test test",
      [RNPaytabsLibrary.city_billing]: "Juffair",
      [RNPaytabsLibrary.state_billing]: "Manama",
      [RNPaytabsLibrary.country_billing]: "BHR",
      [RNPaytabsLibrary.postal_code_billing]: "00973", //Put Country Phone code if Postal code not available '00973'//
      [RNPaytabsLibrary.address_shipping]: "test test",
      [RNPaytabsLibrary.city_shipping]: "Juffair",
      [RNPaytabsLibrary.state_shipping]: "Manama",
      [RNPaytabsLibrary.country_shipping]: "BHR",
      [RNPaytabsLibrary.postal_code_shipping]: "00973", //Put Country Phone code if Postal
      [RNPaytabsLibrary.color]: "#cccccc",
      [RNPaytabsLibrary.theme_light]: false,
      [RNPaytabsLibrary.language]: 'en', // 'en', 'ar'
      [RNPaytabsLibrary.tokenization]: true
    }, (response) => {
      // Callback for success & fail.
    
      // { pt_token_customer_email: '',pt_token: '',pt_token_customer_password: '', pt_transaction_id: '123456',pt_response_code: '100' }
    
      RNPaytabsLibrary.log("on Response Payment");
      console.log(response);
      // Response Code: 100 successful otherwise fail
      if (response.pt_response_code == '100')
        RNPaytabsLibrary.log("Transaction Id: " + response.pt_transaction_id);
      else
        RNPaytabsLibrary.log("Otherwise Response: " + response.pt_response_code);
    
      this.state = { transactionID: response.pt_transaction_id };
    
      // Tokenization
      //RNPaytabs.log(response.pt_token_customer_email);
      //RNPaytabs.log(response.pt_token_customer_password);
      //RNPaytabs.log(response.pt_token);
    
    });
```

**Comment by arhmnsh** [comment]
I still get the same error brother
<img width="1395" alt="image" src="https://user-images.githubusercontent.com/1363724/64114286-47353000-cd95-11e9-8ebd-ef4b7c6766b2.png">

**Comment by RamyHegazy** [comment]
@arhmnsh I am not sure what exactly the issue, i downloaded the repo and tried the example, the only issue i faced was that `npm install` not working fine, has a strange issue, i had to solved:

```bash
$ rm -rf node_modules
$ yarn install 
$ react-native link
$ react-native run-android
```
https://streamable.com/tsl76
